platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/datasource/PathHashRecord.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.virtualmap.datasource;
+
+import com.swirlds.common.crypto.Hash;
+
+/**
+ * A record that contains a path and a hash. It serves for both node types, internal and leaf.
+ * @param path the path of the node
+ * @param hash the hash of the node
+ */
+public record PathHashRecord(long path, Hash hash) {
```

Comment:
Relates to changing it to a record. There is a ticket about making `VirtualRecord` immutable (it has a few setters today, like `setPath()` or `setHash()`). With Java records, it would not be possible to implement, so can be just closed.